### DISCUSSION TITLE
API route to create directory

### DISCUSSION THREAD

**Discussion by Glenn-Mendonca** [coddingtonbear/obsidian-local-rest-api#63] [Ideas]
It would be great if there was a route to create directory in the vault.
 - Reursive directory creation would be extremely great.

Api endpoint structure
```bash
PUT /rootpath HTTP/1.1
Content-Type: text
Body: directory-path-structure | singledirectory-name
```

**Comment by Glenn-Mendonca** [comment]
Can i work upon this idea and create a pull request?

**Reply by coddingtonbear** [reply]
You may want to be aware that the way this library works doesn't involve actually accessing the filesystem directly, and that comes with certain quirks including this: https://github.com/coddingtonbear/obsidian-local-rest-api/issues/21#issuecomment-1140363848.  Without some fairly significant changes to what APIs we use, I'm not totally sure how you might even approach this in a way that would make sense. 

I have to say, though, I'm curious about what your use case is for a feature like this, and wonder if maybe just chipping in on #44 might solve it?

**Comment by Glenn-Mendonca** [comment]
I have an idea for a note-taking browser extension in mind and would like to create folders for organizing notes. However, you are correct that once the #44 pull request is merged, it should fulfill my use case. I apologize for not thoroughly checking the open pull requests and issues, and I appreciate your time. Thank you.
